source

```
@@ -91085,20 +91090,43 @@ document.body.appendChild(frame)</pre>
         <ol>
          <li><p>Perform <span
          data-x="the-hostensurecancompilestrings-implementation">HostEnsureCanCompileStrings</span>(<var>callerRealm</var>,
-         <var>calleeRealm</var>). If this throws an exception, catch it, and <span>report the
-         exception</span>.</p></li>
+         <var>calleeRealm</var>). If this throws an exception, catch it, <span>report the
+         exception</span>, and abort these steps.</p></li>
```

return?

---

This is inside a set of substeps that run in parallel.

---

How can you report an exception from there?

---

Sorry, they run inside a task that is enqueued from in parallel. See https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#timer-initialisation-steps for the actual context.

---

In that case return seems fine, since it's a synchronous algorithm, but okay either way.